File: typescript/hamming/hamming.ts
export function compute(left: string, right: string): number {
  if (left.length !== right.length) {
    throw new Error('DNA strands must be of equal length.')
  }

  return left
          .split('')
          .filter((_, i) => left[i] !== right[i])
          .length
}

File: go/hamming/hamming.go
// Package hamming provides a function the compute distance between dna strands
package hamming

import "fmt"

// Distance computes the distance between two strands
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, fmt.Errorf("Strands %s and %s are not of the same length", a, b)
	}

	var difference int
	for i := range a {
		if a[i] != b[i] {
			difference++
		}
	}
	return difference, nil
}
